src/software/backend/wifi_backend.cpp

```diff
@@ -30,18 +32,15 @@ void WifiBackend::onValueReceived(ConstPrimitiveVectorPtr primitives_ptr)
     primitive_output->sendProto(*createPrimitiveMsg(primitives_ptr));
 }
 
-void WifiBackend::receiveWorld(World world)
+void WifiBackend::onValueReceived(World world)
 {
     vision_output->sendProto(*createVisionMsg(world));
-    Subject<World>::sendValueToObservers(world);
 }
 
-void WifiBackend::receiveTbotsRobotMsg(TbotsRobotMsg robot_msg)
+void WifiBackend::receiveWorld(World world)
 {
-    SensorMsg sensor_msg;
-    TbotsRobotMsg* added_robot_msg = sensor_msg.add_tbots_robot_msg();
-    *added_robot_msg               = robot_msg;
-    Subject<SensorMsg>::sendValueToObservers(sensor_msg);
+    vision_output->sendProto(*createVisionMsg(world));
+    Subject<World>::sendValueToObservers(world);
```

Comment:
Is this for compatibility reasons? according to the sensor fusion rfc, the backend should _not_ be a subject of the world?

Reply:
that's right, we don't want to break anything so we support both interfaces at the same time